Content.Server/GameObjects/Components/Mobs/MobComponent.cs

```diff
@@ -116,42 +164,94 @@ private void ChangeDamageState(ThresholdType threshold)
         }
 
+        public void OnUpdate()
+        {
+            Body.Life(_lifeTick);
+            _lifeTick++;
+        }
+
+
         void IExAct.OnExplosion(ExplosionEventArgs eventArgs)
         {
             var burnDamage = 0;
             var bruteDamage = 0;
             switch(eventArgs.Severity)
             {
                 case ExplosionSeverity.Destruction:
-                    bruteDamage += 250;
-                    burnDamage += 250;
+                    DestroyOwner();
                     break;
                 case ExplosionSeverity.Heavy:
-                    bruteDamage += 60;
-                    burnDamage += 60;
+                    if (_seed.Prob(0.6f))
+                    {
+                        DestroyOwner();
+                    }
+                    else
+                    {
+                        bruteDamage += 60;
+                        burnDamage += 60;
+                    }
                     break;
                 case ExplosionSeverity.Light:
                     bruteDamage += 30;
                     break;
             }
-            Owner.GetComponent<DamageableComponent>().TakeDamage(Shared.GameObjects.DamageType.Brute, bruteDamage);
-            Owner.GetComponent<DamageableComponent>().TakeDamage(Shared.GameObjects.DamageType.Heat, burnDamage);
+            if (bruteDamage > 0)
+            {
+                Owner.GetComponent<DamageableComponent>().TakeDamage(DamageType.Brute, bruteDamage);
+            }
+            if (burnDamage > 0)
+            {
+                Owner.GetComponent<DamageableComponent>().TakeDamage(DamageType.Heat, burnDamage);
+            }
+        }
+
+        private void DestroyOwner()
+        {
+            if (Body != null)
+            {
+                Body.Gib();
+            }
+            var entityManager = IoCManager.Resolve<IEntityManager>();
+            var ghost = entityManager.ForceSpawnEntityAt("MobObserver", Owner.Transform.GridPosition);
+            var mind = Owner.GetComponent<MindComponent>().Mind;
+            mind.UnVisit();
+            mind.Visit(ghost);
+            Owner.Delete();
         }
     }
 
     /// <summary>
-    ///     Fired when <see cref="SpeciesComponent.CurrentDamageState"/> changes.
+    ///     Fired when <see cref="MobComponent.CurrentDamageState"/> changes.
     /// </summary>
     public sealed class MobDamageStateChangedMessage : EntitySystemMessage
     {
-        public MobDamageStateChangedMessage(SpeciesComponent species)
+        public MobDamageStateChangedMessage(MobComponent mob)
         {
-            Species = species;
+            Mob = mob;
         }
 
         /// <summary>
-        ///     The species component that was changed.
+        ///     The mob component that was changed.
         /// </summary>
-        public SpeciesComponent Species { get; }
+        public MobComponent Mob { get; }
+    }
+    [Serializable]
+    public enum AttackTargetDef
```

Comment:
Honestly, looking at this, I'm not sure we really want this kind of targeting system. Literally only one or two zones matter for anything except surgery, and I don't think this system is fundamental enough to deserve being there purely for surgery.

Reply:
#266